internal/types/array.go

```diff
@@ -149,3 +151,50 @@ func (a *Array) Max() any {
 
 	return max
 }
+
+// Contains checks if the Array contains the given value.
+func (a *Array) Contains(filterValue any) bool {
+	switch filterValue := filterValue.(type) {
+	case *Document, *Array:
+		// filterValue is a composite type, so either a and filterValue must be equal
+		// or at least one element of a must be equal with filterValue.
+		// TODO: Compare might be inaccurate for some corner cases, we might want to fix it later.
```

Comment:
If I knew the corner cases, I would cover them in this task. The problem is that I couldn't find them. So, creating a task is not reasonable as it's not straightforward what the task should be about and how much time it would take.

Reply:
> If I knew the corner cases, I would cover them right here, not in a separate issue. 

I would suggest adding the word `unknown` to corner cases.